numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2132,37 +2132,56 @@ array_scalar(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kwds)
     return ret;
 }
 
-static PyObject *
-array_zeros(PyObject *NPY_UNUSED(ignored),
-        PyObject *const *args, Py_ssize_t len_args, PyObject *kwnames)
+HPyDef_METH(array_zeros, "zeros", array_zeros_impl, HPyFunc_KEYWORDS)
+static HPy
+array_zeros_impl(HPyContext *ctx, HPy NPY_UNUSED(ignored), HPy *args, HPy_ssize_t nargs, HPy kw)
```

Comment:
Hmm, we should probably have `const HPy *args` in the signature, to match CPython...